framework/Distributions.py

```diff
@@ -1704,6 +1708,133 @@ def rvs(self):
 
 DistributionsCollection.addSub(Categorical.getInputSpecification())
 
+class MarkovCategorical(Categorical):
+  """
+    Class for the Markov categorical distribution based on "Markov Model"
+    Note: this distribution can have only numerical (float) outcome; in the future we might want to include also the possibility to give symbolic outcome
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = InputData.parameterInputFactory(cls.__name__, ordered=True, baseNode=None)
+
+    StatePartInput = InputData.parameterInputFactory("state", contentType=InputData.StringType)
+    StatePartInput.addParam("outcome", InputData.FloatType, True)
+    StatePartInput.addParam("index", InputData.IntegerType, True)
+    TransitionInput = InputData.parameterInputFactory("transition", contentType=InputData.StringType)
+    inputSpecification.addSub(StatePartInput, InputData.Quantity.one_to_infinity)
+    inputSpecification.addSub(TransitionInput, InputData.Quantity.zero_to_one)
+    inputSpecification.addSub(InputData.parameterInputFactory("workingDir", contentType=InputData.StringType))
+    ## Because we do not inherit from the base class, we need to manually
+    ## add the name back in.
+    inputSpecification.addParam("name", InputData.StringType, True)
+
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Function that initializes the categorical distribution
+      @ In, None
+      @ Out, none
+    """
+    Categorical.__init__(self)
+    self.dimensionality = 1
+    self.disttype       = 'Discrete'
+    self.type           = 'MarkovCategorical'
+    self.steadyStatePb  = None
+    self.transition     = None
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the distribution parameter input.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    workingDir = paramInput.findFirst('workingDir')
+    if workingDir is not None:
+      self.workingDir = workingDir.value
+    else:
+      self.workingDir = os.getcwd()
```

Comment:
mmmm...is workingDir required? If so, you should not have this else statement

Reply:
no, it is not required